k8sutils/pkg/cri/criwrapper.go

```diff
@@ -0,0 +1,180 @@
+package criwrapper
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+	odigosv1 "github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	criapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+type CriClient struct {
+	conn   *grpc.ClientConn
+	client criapi.RuntimeServiceClient
+	Logger logr.Logger
+}
+
+// Default runtime endpoints
+var defaultRuntimeEndpoints = []string{
+	"unix:///run/containerd/containerd.sock",
+	"unix:///run/crio/crio.sock",
+	"unix:///var/run/cri-dockerd.sock",
+}
+
+func detectRuntimeSocket() string {
+	for _, endpoint := range defaultRuntimeEndpoints {
+		// Extract the file path from the endpoint
+		socketPath := strings.TrimPrefix(endpoint, "unix://")
+		if _, err := os.Stat(socketPath); err == nil {
+			return endpoint
+		}
+	}
+	return ""
+}
+
+// Connect attempts to establish a connection to a CRI runtime.
+func (rc *CriClient) Connect(ctx context.Context) error {
+	var err error
+
+	endpoint := detectRuntimeSocket()
+	if endpoint == "" {
+		return fmt.Errorf("unable to detect CRI runtime endpoint")
```

Comment:
nit: (as the error doesn't include formatting)
```suggestion
		return errors.New("unable to detect CRI runtime endpoint")
```

We could also export this by declaring above a package level error
```Go
var ErrDetectingCRI = errors.New("unable to detect CRI runtime endpoint")
```
This will allow the caller of this function hande this specific error